### PR TITLE
Add plantlike leaves

### PR DESCRIPTION
**Pull request by numberZero**
Some mods (e.g. Ethereal) already use the plantlike drawtype for the leaves because that looks better than even the fancy leaves.

### CODE REVIEW

File: src/nodedef.cpp [hunk 1]
```diff
@@ -633,7 +635,14 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 		visual_solidness = 1;
 		break;
 	case NDT_ALLFACES_OPTIONAL:
-		if (tsettings.leaves_style == LEAVES_FANCY) {
+		if (tsettings.leaves_style == LEAVES_PLANTLIKE) {
+			drawtype = NDT_PLANTLIKE;
+			solidness = 0;
+			visual_scale = 1.2;
```
Comment (paramat) [inline src/nodedef.cpp]: I guess this is to make plantlike horizontally fill the diagonal, but textures will vertically extend beyond node edges, and Z-fight with other leaves, so not sure about this.

**Comment by paramat** [comment]
One problem with plantlike is that leaves become invisible from above, i find it looks worse than the other options while being as intensive as fancy.
:-1: I would rather this was done in a subgame by defining a node as plantlike, then they can customise the texture to not extend so far vertically.

File: src/nodedef.cpp (hunk 1 above)
Comment (numberZero) [inline src/nodedef.cpp]: They don’t Z-fight as they are not strictly diagonal (1° off IIRC)

Comment (paramat) [inline src/nodedef.cpp]: Horizontally yes, but vertically they will.

**Comment by numberZero** [comment]
But I consider plantlike leaves nicer than others. What’s about adding this as an option, keeping fancy leaves default?

**Comment by paramat** [comment]
Better to use a dedicated node and texture in a mod or subgame. The standard square texture is not suitable for stretching diagonally across a node.

**Comment by numberZero** [comment]
@paramat Then let it be `plantlike-optional`. Don't force mod makers to use slow `plantlike` drawtype for leaves as they currently do to achieve better look.

**Comment by paramat** [comment]
Plantlike drawtype isn't slow. But yes a new engine drawtype would be another way, that automatically stretches a square texture over the diagonal. However i still don't feel it's important enough to do and it can already be done with plantlike, so i don't agree to a new drawtype.

**Comment by numberZero** [comment]
@paramat You’re right that this should be done in a mod... Ethereal not only changes the drawtype, it also tweaks leaves making them non-walkable, and that looks very well. But *walkable* plantlike leaves are really weird... Closing this.

**Pull request closed without merging**